### ISSUE TITLE
Feature Request: `AWS::StepFunctions::StateMachine` - (JSONPath) Recognize in a `choice` state that the `Variable` needs to start with "$"

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4076]
### Describe the feature you'd like to request

Currently cfn-lint is unable to recognize if the `Variable` in a `choice` state starts with "$". The following CloudFormation Template will cause a CloudFormation error:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Pass
        States:
          Pass:
            Type: Pass
            Next: Choice
          Choice:
            Type: Choice
            Choices:
              - Variable: "Test" # <------------------------ Needs to start with "$"
                StringMatches: "Test"
                Next: Success
            Default: Success
          Success:
            Type: Succeed
```

CloudFormation error:

```
Resource handler returned message:
"Invalid State Machine Definition: 'SCHEMA_VALIDATION_FAILED:
Value is not a Reference Path: Reference path didn't start with '$' at /States/Choice/Choices[0]/Variable'
(Service: Sfn, Status Code: 400, Request ID: 6...3)
(SDK Attempt Count: 1)"
(RequestToken: 4...0, HandlerErrorCode: InvalidRequest)
```

It would be great if cfn-lint could give an error on this.

### Describe the solution you'd like

A new/updated rule to notify the user of this error.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
